src/openforms/variables/models.py

```diff
@@ -119,12 +124,13 @@ def request_arguments(self, context: DataMapping) -> dict:
         # this catches faults requests doesn't: https://github.com/psf/requests/issues/6359
         HeaderValidator()(headers)
 
-        escaped_for_query = {k: quote(str(v)) for k, v in context.items()}
         escaped_for_path = {k: quote_plus(str(v)) for k, v in context.items()}
 
-        query_params = self.query_params.format(**escaped_for_query)
-        query_params = (
-            query_params[1:] if query_params.startswith("?") else query_params
+        query_params = urlencode(
+            {
+                param: value.format(**context)
+                for param, value in (self.query_params or {}).items()
+            }
         )
```

Comment:
afaik we don't even need the `urlencode` because you can pass a dict for `params` to requests, see https://requests.readthedocs.io/en/latest/api/#requests.request

so this would be sufficient:

```py
query_params = {param: value.format(**context) for param, value in (self.query_params or {}).items()}
```